### PR TITLE
Pyre Unit Tests

### PR DESCRIPTION
**Pull request by gtw2**
Adds unit tests, default values, and cleanup

### CODE REVIEW

**Comment by gtw2** [comment]
Nosetests are out of date, and will be fixed with @jbae11 's PR I believe

File: src/pyre_tests.cc [hunk 1]
```diff
@@ -1,47 +1,391 @@
...
+      "<leftover_commod>waste</leftover_commod>"
+      "<throughput>100</throughput>"
+      "<feedbuf_size>100</feedbuf_size>"
+      "<feed_commods> <val>feed</val> </feed_commods>"
+  ;
+  
+  
+  cyclus::MockSim sim1(cyclus::AgentSpec(":recycle:Pyre"), config, simdur);
+  
+  EXPECT_NO_THROW(sim1.Run()) << "Cumulative separation efficiency smaler than 1 is throwing an error but should not.";
```
Comment (katyhuff) [inline src/pyre_tests.cc]:
```suggestion
  EXPECT_NO_THROW(sim1.Run()) << "Cumulative separation efficiency < 1 should not throw an error.";
```

File: src/pyre_tests.cc [hunk 2]
```diff
@@ -1,47 +1,391 @@
...
+      "<leftover_commod>waste</leftover_commod>"
+      "<throughput>100</throughput>"
+      "<feedbuf_size>100</feedbuf_size>"
+      "<feed_commods> <val>feed</val> </feed_commods>"
+      ;
+  
+  
+  cyclus::MockSim sim2(cyclus::AgentSpec(":recycle:Pyre"), config, simdur);
+  
+  EXPECT_NO_THROW(sim2.Run()) << "Cumulative separation efficiency of 1 is throwing an error but should not.";
```
Comment (katyhuff) [inline src/pyre_tests.cc]:
```suggestion
  EXPECT_NO_THROW(sim2.Run()) << "Cumulative separation efficiency = 1 should not throw an error.";
```

File: src/pyre_tests.cc [hunk 3]
```diff
@@ -1,47 +1,391 @@
...
+      ""
+      "<leftover_commod>waste</leftover_commod>"
+      "<throughput>100</throughput>"
+      "<feedbuf_size>100</feedbuf_size>"
+      "<feed_commods> <val>feed</val> </feed_commods>"
+      ;
+
+  cyclus::MockSim sim1(cyclus::AgentSpec(":recycle:Pyre"), config, simdur);
+  
+  EXPECT_THROW(sim1.Run(), cyclus::ValueError) << "Direct separation efficiency greater than 1 is not throwing an error but should.";
```
Comment (katyhuff) [inline src/pyre_tests.cc]:
```suggestion
  EXPECT_THROW(sim1.Run(), cyclus::ValueError) << "Direct separation efficiency > 1 should throw an error.";
```

File: src/pyre_tests.cc [hunk 4]
```diff
@@ -1,47 +1,391 @@
...
+    ""
+    "<leftover_commod>waste</leftover_commod>"
+    "<throughput>100</throughput>"
+    "<feedbuf_size>100</feedbuf_size>"
+    "<feed_commods> <val>feed</val> </feed_commods>"
+    ;
+  
+  cyclus::MockSim sim2(cyclus::AgentSpec(":recycle:Pyre"), config, simdur);
+  
+  EXPECT_THROW(sim2.Run(), cyclus::ValueError) << "Single cumulative separation efficiency greater than 1 is not throwing an error but should.";
```
Comment (katyhuff) [inline src/pyre_tests.cc]:
```suggestion
  EXPECT_THROW(sim2.Run(), cyclus::ValueError) << "Single cumulative sep efficiency > 1 should throw an error.";
```

File: src/pyre_tests.cc [hunk 5]
```diff
@@ -1,47 +1,391 @@
...
+      ""
+      "<leftover_commod>waste</leftover_commod>"
+      "<throughput>100</throughput>"
+      "<feedbuf_size>100</feedbuf_size>"
+      "<feed_commods> <val>feed</val> </feed_commods>"
+      ;
+  
+  cyclus::MockSim sim3(cyclus::AgentSpec(":recycle:Pyre"), config, simdur);
+  
+  EXPECT_THROW(sim3.Run(), cyclus::ValueError) << "Multiple cumulative separation efficiencies greater than 1 are not throwing an error but should.";
```
Comment (katyhuff) [inline src/pyre_tests.cc]: as above, shorten.

File: src/pyre_tests.cc [hunk 6]
```diff
@@ -1,47 +1,391 @@
...
+  int id = sim.Run();
 
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-// Do Not Touch! Below section required for connection with Cyclus
-cyclus::Agent* PyreConstructor(cyclus::Context* ctx) {
-  return new Pyre(ctx);
+  std::vector<Cond> conds;
+  conds.push_back(Cond("SenderId", "==", id));
+  int resid = sim.db().Query("Transactions", &conds).GetVal<int>("ResourceId");
+  MatQuery mq (sim.GetMaterial(resid));
```
Comment (katyhuff) [inline src/pyre_tests.cc]:
```suggestion
  MatQuery mq (sim.GetMaterial(resid));
  default_efficiency = 0.7156278629279868703;
```

File: src/pyre_tests.cc [hunk 7]
```diff
@@ -1,47 +1,391 @@
...
+  EXPECT_DOUBLE_EQ(m[922350000]*0.6*0.7156278629279868703*100, mq.mass("U235"));
```
Comment (katyhuff) [inline src/pyre_tests.cc]:
```suggestion
  EXPECT_DOUBLE_EQ(m[922350000]*0.6*default_efficiency*100, mq.mass("U235"));
```

File: src/pyre_tests.cc [hunk 8]
```diff
@@ -1,47 +1,391 @@
...
+  EXPECT_DOUBLE_EQ(m[922380000]*0.6*0.7156278629279868703*100, mq.mass("U238"));
```
Comment (katyhuff) [inline src/pyre_tests.cc]:
```suggestion
  EXPECT_DOUBLE_EQ(m[922380000]*0.6*default_efficiency*100, mq.mass("U238"));
```

File: src/pyre_tests.cc [hunk 9]
```diff
@@ -1,47 +1,391 @@
...
+  EXPECT_DOUBLE_EQ(m[942390000]*0.7*0.7156278629279868703*100, mq.mass("Pu239"));
```
Comment (katyhuff) [inline src/pyre_tests.cc]:
```suggestion
  EXPECT_DOUBLE_EQ(m[942390000]*0.7*default_efficiency*100, mq.mass("Pu239"));
```

File: src/pyre_tests.cc (hunk 6 above)
Comment (katyhuff) [reply src/pyre_tests.cc]: (also maybe add an inline comment explaining this number (wherefore?)

File: src/pyre_tests.cc [hunk 10]
```diff
@@ -1,47 +1,391 @@
...
-class PyreTest : public ::testing::Test {
- protected:
-  cyclus::TestContext tc;
-  Pyre* facility;
+namespace recycle {
+  
+// Check that cumulative separations efficiency for a single nuclide of less than or equal to one does not trigger an error.
+TEST(PyreTests, SeparationEfficiency) {
 
-  virtual void SetUp() {
```
Comment (katyhuff) [inline src/pyre_tests.cc]: why no more setup?

File: src/pyre_tests.cc [hunk 11]
```diff
@@ -1,47 +1,391 @@
...
+  sim.AddSource("feed").recipe("recipe1").Finalize();
+  sim.AddSink("volox_stream").capacity(100).Finalize();
+  sim.AddRecipe("recipe1", c);
+  int id = sim.Run();
+
+  QueryResult qr = sim.db().Query("AgentPosition", NULL);
+  EXPECT_EQ(qr.GetVal<double>("Latitude"), 10.0);
+  EXPECT_EQ(qr.GetVal<double>("Longitude"), 15.0);
+ }
+} // namespace cycamore
```
Comment (katyhuff) [inline src/pyre_tests.cc]: cycamore -> recycle

**Comment by gtw2** [comment]
@katyhuff can this be merged now? - I'm leaving the setup and config string changes we talked about for a different PR.

**Comment by katyhuff** [comment]
looks good.

**Pull request merged**